noetic/Dockerfile

```diff
@@ -0,0 +1,142 @@
+# CARMA Base Image Docker Configuration Script
+
+FROM nvidia/cudagl:11.2.0-devel-ubuntu20.04
+
+ARG DOCKERFILE_DIR="noetic"
+
+ENV NVIDIA_VISIBLE_DEVICES=${NVIDIA_VISIBLE_DEVICES:-all}
+ENV NVIDIA_DRIVER_CAPABILITIES=${NVIDIA_DRIVER_CAPABILITIES:+$NVIDIA_DRIVER_CAPABILITIES,}graphics
+
+ARG DEBIAN_FRONTEND="noninteractive"
+
+# Step 1: Set up the carma user
+RUN useradd -m carma && \
+    echo "carma:carma" | chpasswd && \
+    usermod --shell /bin/bash carma && \
+    usermod -aG sudo carma && \
+    mkdir -p /etc/sudoers.d && \
+    echo "carma ALL=(ALL) NOPASSWD:ALL" >> /etc/sudoers.d/carma && \
+    chmod 0440 /etc/sudoers.d/carma && \
+    usermod  --uid 1000 carma && \
+    groupmod --gid 1000 carma && \
+    mkdir -p /opt/carma/launch /opt/carma/logs /opt/carma/routes && \
+    chown carma:carma -R /opt/carma
+
+# Copy dependency files
+COPY ${DOCKERFILE_DIR}/base_deps.txt ${DOCKERFILE_DIR}/additional_deps.txt ${DOCKERFILE_DIR}/ros_deps.txt /tmp/
+
+# Step 2: Install base dependencies
+RUN apt-get update && \
+    xargs -a /tmp/base_deps.txt apt-get install --no-install-recommends -y && \
+    apt-get clean && \
+    rm -rf /var/lib/apt/lists/*
+
+# Step 3: Install additional dependencies
+RUN apt-get update && \
+    xargs -a /tmp/additional_deps.txt apt-get install --no-install-recommends -y && \
+    apt-get clean && \
+    rm -rf /var/lib/apt/lists/*
+
+# Step 4: Set up ROS repository and install ROS dependencies
+ARG ROS_DISTRO=noetic
+RUN apt-key adv --keyserver 'hkp://keyserver.ubuntu.com:80' --recv-key C1CF6E31E6BADE8868B172B4F42ED6FBAB17C654 && \
+    echo "deb http://packages.ros.org/ros/ubuntu $(lsb_release -sc) main" > /etc/apt/sources.list.d/ros-latest.list && \
+    apt-get update && \
+    xargs -a /tmp/ros_deps.txt apt-get install --no-install-recommends -y && \
+    apt-get clean && \
+    rm -rf /var/lib/apt/lists/*
+
+# Step 5: Install AutonomouStuff dependencies
+RUN echo "deb [trusted=yes] https://s3.amazonaws.com/autonomoustuff-repo/ $(lsb_release -sc) main" > /etc/apt/sources.list.d/autonomoustuff-public.list && \
+    apt-get update && \
+    apt-get install --no-install-recommends -y libas-common && \
+    apt-get clean && \
+    rm -rf /var/lib/apt/lists/*
+
+# Step 6: Install Vimba dependencies
+RUN add-apt-repository --update -y ppa:rock-core/qt4 && \
```

Comment:
Just want to check if we can combine these multiple RUN statements ? I think we were trying to reduced previously for our final base image size. I agree with having only single RUN is harder. Do we want to keep them split or merge few of them?

Reply:
Decided make the optimization for later